### PR TITLE
Closure all the things

### PR DESCRIPTION
**Pull request by kripken**
Followup to #728 

As with there, this should be safe due to using MODULARIZE everywhere (nice job!).

### CODE REVIEW

**Comment by surma** [comment]
We actually need to run the build script as we commit the wasm binaries. If you don‘t have the time, I can do this tomorrow or so :)

File: codecs/imagequant/build.sh [hunk 1]
```diff
@@ -14,6 +14,7 @@ echo "============================================="
   emcc \
     --bind \
     ${OPTIMIZE} \
+    --closure 1 \
```
Comment (surma) [inline codecs/imagequant/build.sh]: We don’t need this here as we are just generating object files

Comment (kripken) [reply codecs/imagequant/build.sh]: Oh right, there's a `-c` there... Fixed and I removed some other unnecessary flags as well.

**Comment by kripken** [comment]
If it's easy for you to build that would be good - I managed to build one of them but I had to fiddle with the Makefile for a while (something odd with the prefix on my system).

**Comment by RReverser** [comment]
@kripken If you use `npm run build:image` + `npm run build`, it would build codecs within Docker containers, so system prefixes shouldn't matter.

**Comment by kripken** [comment]
@RReverser do I need to do any special setup for that (aside from `npm install`)? I get
```
$ npm run build:image
npm ERR! missing script: build:image

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/azakai/.npm/_logs/2020-04-15T20_03_09_336Z-debug.log
```
Nothing much useful in the log, ends in
```
verbose node v12.4.0
verbose npm  v6.9.0
error missing script: build:image
```

**Comment by RReverser** [comment]
For packages that don't have `build:image`, just `build` should suffice.

**Comment by kripken** [comment]
Oh ok, thanks!

Updated now. Here are the size differences:
```
Before:

  63554  codecs/imagequant/imagequant.js
  68064  codecs/imagequant/imagequant.wasm
  52781  codecs/mozjpeg_enc/mozjpeg_enc.js
 241369  codecs/mozjpeg_enc/mozjpeg_enc.wasm
  60476  codecs/webp_dec/webp_dec.js
 100224  codecs/webp_dec/webp_dec.wasm
  50200  codecs/webp_enc/webp_enc.js
 175751  codecs/webp_enc/webp_enc.wasm

After:

  26964  codecs/imagequant/imagequant.js
  64044  codecs/imagequant/imagequant.wasm
  22024  codecs/mozjpeg_enc/mozjpeg_enc.js
 231072  codecs/mozjpeg_enc/mozjpeg_enc.wasm
  26217  codecs/webp_dec/webp_dec.js
  96722  codecs/webp_dec/webp_dec.wasm
  21636  codecs/webp_enc/webp_enc.js
 170766  codecs/webp_enc/webp_enc.wasm
```
JS files shrank by over 50% as expected, but the wasm files shrank a little too, I guess due to using a newer emscripten version?

**Comment by RReverser** [comment]
Yeah could be. I wonder what gzip and/or brotli differences look like for JS.

**Comment by jakearchibald** [comment]
> I wonder what gzip and/or brotli differences look like for JS.

The ci reports this:

```
Build change report:
  CHANGED build/report.html - 84 kB -> 84 kB (-1 B, 1x)
  CHANGED build/serviceworker.js - 2.22 kB -> 2.22 kB (-4 B, 0.998x)
  CHANGED build/imagequant.5a086.wasm - 31.8 kB -> 29.9 kB (-1.95 kB, 0.939x)
    Renamed from: build/imagequant.c970f.wasm
  CHANGED build/main-app.eba7d.js - 22.2 kB -> no change
    Renamed from: build/main-app.bc3ef.js
  CHANGED build/mozjpeg_enc.2d0c9.wasm - 57.6 kB -> 53 kB (-4.69 kB, 0.919x)
    Renamed from: build/mozjpeg_enc.93395.wasm
  CHANGED build/process-imagequant.86d82.worker.js - 13 kB -> 9.71 kB (-3.28 kB, 0.747x)
    Renamed from: build/process-imagequant.75bc9.worker.js
  CHANGED build/process-mozjpeg-enc.edfe9.worker.js - 11.4 kB -> 8.29 kB (-3.07 kB, 0.73x)
    Renamed from: build/process-mozjpeg-enc.fc3de.worker.js
  CHANGED build/process-webp-dec.e3073.worker.js - 12.4 kB -> 9.51 kB (-2.93 kB, 0.764x)
    Renamed from: build/process-webp-dec.df90a.worker.js
  CHANGED build/process-webp-enc.ff08b.worker.js - 11.1 kB -> 7.91 kB (-3.14 kB, 0.716x)
    Renamed from: build/process-webp-enc.59e06.worker.js
  CHANGED build/processor-worker.de0bf.worker.js - 3.1 kB -> 3.1 kB (-1 B, 1x)
    Renamed from: build/processor-worker.d7404.worker.js
  CHANGED build/webp_dec.557c3.wasm - 44.4 kB -> 42.6 kB (-1.81 kB, 0.959x)
    Renamed from: build/webp_dec.fa0ab.wasm
  CHANGED build/webp_enc.ed3ae.wasm - 82.9 kB -> 80.3 kB (-2.62 kB, 0.968x)
    Renamed from: build/webp_enc.ea665.wasm
```

**Comment by RReverser** [comment]
@jakearchibald Awesome, thanks!

**Comment by surma** [comment]
Aw man, I forgot we had this. I should have checked this before tweeting.

**Comment by RReverser** [comment]
Add a follow-up tweet to the thread perhaps? Not ideal, but it will still get visibility with two tweets shown together in the feed for most people.

**Pull request merged**